### ISSUE TITLE
ImageLayout error on ops.Erase

### ISSUE THREAD

**Issue by bhheo** [NVIDIA/DALI#2126]
Hi

I'm working with DALI 0.23.0 on ImageNet dataset
I get an error when using ops.CropMirrorNormalize after ops.Erase

The error message is
[/opt/dali/dali/operators/image/crop/crop_mirror_normalize.h:166] Assert on "ImageLayoutInfo::IsImage(input_layout_)" failed: Unsupported layout: '' for input 0 '__Erase_3'

Here is the code can reproduce error above
```Python
class HybridTrainPipe(Pipeline):
    def __init__(self, batch_size, num_threads, device_id, num_gpus, opt,
                 file_root, image_size=224, file_list=""):
        super(HybridTrainPipe, self).__init__(batch_size, num_threads,
                                              device_id, seed=12 + device_id)
        self.input = ops.FileReader(file_root=file_root,
                                    file_list=file_list,
                                    shuffle_after_epoch=True,
                                    shard_id=device_id,
                                    num_shards=num_gpus)
        self.decode_rrc = ops.ImageDecoderRandomCrop(
            device="cpu",
            output_type=dali.types.RGB,
            random_area=[opt.rrc_minimum, 1])
        self.res = ops.Resize(device="gpu",
                              resize_x=image_size, resize_y=image_size,
                              interp_type=dali.types.INTERP_LINEAR)
        self.erase = ops.Erase(device="gpu",
                               axis_names="HW",
                               anchor=(40,20),
                               shape=(10,10))
        self.cmnp = ops.CropMirrorNormalize(device="gpu",
                                            output_dtype=dali.types.FLOAT,
                                            output_layout=dali.types.NCHW,
                                            image_type=dali.types.RGB)

    def define_graph(self):
        self.jpegs, self.labels = self.input(name="Reader")
        images = self.decode_rrc(self.jpegs).gpu()
        images = self.res(images)
        images = self.erase(images)
        output = self.cmnp(images)

        return [output, self.labels]
```

I think it's relevant to #2116 issue. But, the function is different.

**Comment by klecki** [comment]
Hi,
indeed, looking through the Erase source it seems to be the same problem.
As a workaround I can suggest setting the layout manually with [Reshape Op](https://docs.nvidia.com/deeplearning/dali/master-user-guide/docs/supported_ops.html#nvidia.dali.ops.Reshape) as suggested in that issue:

```
def __init__(...):
    ...
    self.reshape = ops.Reshape(device="cpu", layout="HWC")

def define_graph(self):
   ...
   images = self.erase(images)
   images = self.reshape(images) 
   ...
```

We will adjust the operator to fix that.

**Comment by bhheo** [comment]
Thanks.
My code is working well with `self.reshape` trick.
I hope this bug will be fixed in the next version.

**Comment by klecki** [comment]
Hi, the bug will be fixed in #2133. We will be also checking if it doesn't appear in other operators.
